src/frontend/src/eth/components/transactions/Transaction.svelte

```diff
@@ -35,12 +35,10 @@
 	$: ({ value, timestamp, displayTimestamp, uiType: type } = transaction);
 
 	let ckTokenSymbol: string;
-	$: ckTokenSymbol =
-		nonNullish($tokenId) && isSupportedEthTokenId($tokenId)
-			? // TODO: improve the way we link ETH and SepoliaETH to their twin tokens
-				`ck${$tokenSymbol}`
-			: // TODO: $token could be undefined, that's why we cast as `Erc20Token | undefined`; adjust the cast once we're sure that $token is never undefined
-				($token as Erc20Token | undefined)?.twinTokenSymbol ?? '';
+	$: ckTokenSymbol = isSupportedEthToken($token)
+		? $token.twinTokenSymbol
+		: // TODO: $token could be undefined, that's why we cast as `Erc20Token | undefined`; adjust the cast once we're sure that $token is never undefined
+			($token as Erc20Token | undefined)?.twinTokenSymbol ?? '';
```

Comment:
shouldn't we cast to `TokenWithLinkedData | undefined`?

Reply:
In the code the linked data are included already for ERC20:

`export type Erc20Token = Erc20Contract & Token;`

and 

`export type Erc20Contract = Erc20ContractAddress & { exchange: Exchange } & TokenLinkedData;`

this is used throughout the code, that is why I didn't change it, or moved up the `TokenLinkedData`

Reply:
Gotcha!